tests/run-duration.feature

```diff
@@ -0,0 +1,12 @@
+# Tests for the workflow execution duration time
+
+Feature: Run duration
+
+    As a researcher,
+    I want to verify that my workflow finishes in a reasonable amount of time,
+    so that I can stay assured that there are no unusual problems with computing resources.
+
+    Scenario: The workflow terminates in a reasonable amount of time
+        When the workflow execution completes
+        Then the workflow run duration should be less than 25 minutes
```

Comment:
25 minutes may be too pessimistic? It takes about 40 seconds on my laptop. We could use say 5 minutes or 10 minutes as maximum.